lib/klaviyo/client.rb

```diff
@@ -1,74 +1,96 @@
-require 'open-uri'
-require 'base64'
-require 'json'
-
 module Klaviyo
-  class KlaviyoError < StandardError; end
-  
   class Client
-    def initialize(api_key, url = 'http://a.klaviyo.com/')
-      @api_key = api_key
-      @url = url
-    end
-    
-    def track(event, kwargs = {})
-      defaults = {:id => nil, :email => nil, :properties => {}, :customer_properties => {}, :time => nil}
-      kwargs = defaults.merge(kwargs)
-      
-      if kwargs[:email].to_s.empty? and kwargs[:id].to_s.empty?
-        raise KlaviyoError.new('You must identify a user by email or ID')
+    DOMAIN = 'https://a.klaviyo.com/api'
+    V1 = 'v1'
+    V2 = 'v2'
+
+    HTTP_GET = 'GET'
+    HTTP_POST = 'POST'
+    HTTP_PUT = 'PUT'
+    HTTP_DELETE = 'DELETE'
+
+    ALL = 'all'
+    EXCLUSIONS = 'exclusions'
+    EXPORT = 'export'
+    GROUP = 'group'
+    LIST = 'list'
+    LISTS = 'lists'
+    MEMBERS = 'members'
+    METRIC = 'metric'
+    METRICS = 'metrics'
+    PERSON = 'person'
+    SUBSCRIBE = 'subscribe'
+    TIMELINE = 'timeline'
+
+    DEFAULT_COUNT = 100
+    DEFAULT_PAGE = 0
+    DEFAULT_SORT = 'desc'
```

Comment:
DEFAULT_SORT should include direction in its name so users of the SDK don't need to go to the API docs.